Felienne/hedy
# [FEATURE] Supports multiple teachers per class

**Description**
In this PR we add the functionality for a class to have multiple teachers. As this touches our core code of class functionality a lot of re-writing had to be done. Therefore this PR should be very extensively tested before merging. Shortly the following changes are made:
- Front-end changes are made to support a "Add teacher" button to the class overview
- The `teacher` attribute on the Class table is change to a `teachers` attribute of the set type
- Each teacher user is updated with a new attribute `teacher_classes`
    - This is a set of all the classes of a teacher
    - Necessary to reduce server load and still efficiently retrieve all teacher classes 

Some things that are still open for discussion:
- Currently a user is automatically added as a teacher, is this desirable?
- On account removal all related classes are removed, what if there are multiple teachers?
- Do we want to "lock" customizations when one teacher is working on them?
   - This is really really complex and I propose to not strive to implement this 


**Fixes**
This PR fixes #3197 and fixes #3220.

**How to test**
Make sure you are a teacher and have another teacher account. Verify that all functionality is working as it should (creating, deleting, customizing a class). As well as removing, adding and inviting a student. Add a new teacher and verify that you can only add a user that is actually a teacher. Verify that the other user can log in and also access the class.
